mis/pipeline/pulse.py

```diff
@@ -20,59 +26,119 @@ class BasePulseShaper(ABC):
     is passed at the time of pulse generation, not during initialization.
     """
 
-    def __init__(self, instance: Any, config: SolverConfig):
-        """
-        Initialize the pulse shaping module with a MIS instance.
+    duration_us: int | None = None
+    """The duration of the pulse.
 
-        Args:
-            instance (MISInstance): The MIS problem instance.
-        """
-        self.instance = instance
-        self.config: SolverConfig = config
-        self.pulse: Pulse | None = None
+    If unspecified, use the maximal duration for the device."""
 
     @abstractmethod
-    def generate(self, register: Register) -> Pulse:
+    def generate(self, config: SolverConfig, register: Register, graph: nx.Graph) -> Pulse:
         """
         Generate a pulse based on the problem and the provided register.
 
         Args:
-            register (Register): The physical register layout.
+            config: The configuration for this pulse.
+            register: The physical register layout.
+            graph: The graph for the physical register layout.
```

Comment:
Err, why do you need these two parameters ?

Reply:
Because `DefaultPulseShaper` (see below) needs them both.